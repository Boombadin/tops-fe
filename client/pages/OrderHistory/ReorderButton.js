import { breakpoint } from '@central-tech/core-ui';
import React from 'react';
import styled from 'styled-components';

import withLocales from '@client/hoc/withLocales';
import { Loader } from '@client/magenta-ui';
import {
  ELEMENT_ACTION,
  ELEMENT_TYPE,
  generateTestId,
} from '@client/utils/generateElementId';

import { Context } from './Context';

const Reorder = styled.button`
  width: 100px;
  height: 32px;
  font-size: 13px;
  align-self: center;
  min-width: 80px;
  max-width: 120px;
  padding: 5px 0;
  line-height: 22px;
  font-weight: 700;
  border-radius: 5px;
  border: solid 1px #ed1d23;
  background-color: #ffffff;
  color: #544943;
  cursor: pointer;
  ${breakpoint('xs', 'md')`
    width: 100%;
    height: 36px;
    max-width: none;

  `}
`;

const ReorderButton = ({ translate, orderId, disabled }) => {
  return (
    <Context.Consumer>
      {({ processedOrderId, reorder }) => (
        <Reorder
          data-testid={generateTestId({
            type: ELEMENT_TYPE.BUTTON,
            action: ELEMENT_ACTION.ADD,
            moduleName: 'Reorder',
            uniqueId: `reorder-${orderId}`,
          })}
          onClick={() => reorder(orderId)}
          disabled={disabled}
          color="#544943"
          radius={5}
        >
          {processedOrderId === orderId ? (
            <Loader active inline size="mini" />
          ) : (
            translate('order_history.table.repeat')
          )}
        </Reorder>
      )}
    </Context.Consumer>
  );
};

export default withLocales(ReorderButton);
