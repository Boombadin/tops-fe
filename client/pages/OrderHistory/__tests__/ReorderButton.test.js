import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import ReorderButton from '../ReorderButton';

jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});

const TEST_ID = {
  BUTTON_REORDER: 'btn-addReorder-reorder-621636',
};

describe('features/orderDetail/components/ReorderButton', () => {
  const props = {
    orderId: '621636',
    reorder: () => {},
  };

  test('it should be match snapshot', () => {
    const { asFragment } = render(<ReorderButton {...props} />);
    expect(asFragment(<ReorderButton {...props} />)).toMatchSnapshot();
  });

  test('click reorder process | it should be match snapshot', () => {
    const { getByTestId, asFragment } = render(<ReorderButton {...props} />);
    const buttonViewOrderReorder = getByTestId(TEST_ID.BUTTON_REORDER);
    expect(buttonViewOrderReorder).toBeInTheDocument();
    fireEvent.click(buttonViewOrderReorder);
    expect(asFragment()).toMatchSnapshot();
  });
});
