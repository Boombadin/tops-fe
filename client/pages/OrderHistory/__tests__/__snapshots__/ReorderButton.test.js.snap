// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/orderDetail/components/ReorderButton click reorder process | it should be match snapshot 1`] = `
<DocumentFragment>
  <button
    class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
    color="#544943"
    data-testid="btn-addReorder-reorder-621636"
    radius="5"
  >
    order_history.table.repeat
  </button>
</DocumentFragment>
`;

exports[`features/orderDetail/components/ReorderButton it should be match snapshot 1`] = `
<DocumentFragment>
  <button
    class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
    color="#544943"
    data-testid="btn-addReorder-reorder-621636"
    radius="5"
  >
    order_history.table.repeat
  </button>
</DocumentFragment>
`;
